.github/workflows/main.yml

```
@@ -0,0 +1,33 @@
+name: CI Workflow
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Run ESLint
+        run: npm run lint
+
+      - name: Run Prettier Formatter
+        run: npm run format
```

```suggestion
        run: npm run format:check
```

Sorry I missed this before. But these needs to be changes to something that only _checks_ the format. Otherwise this will always pass because it will just format the files.